nselib/httpcookies.lua

```diff
@@ -0,0 +1,476 @@
+---
+-- The httpcookies library will provide a complete implementation of 
+-- cookies. The library will prove useful to new scripts as we have 
+-- an easy option to manage cookies. Existing scripts will be benefitted
+-- by the library too as the arguments to the library can be easily
+-- passed by scripts using http/httpspider library. The library will also 
+-- be helpful in setting up arbitrary cookie values and sending it in the 
+-- http requests which can prove to be quite useful in running scripts on
+-- sites which authenticates through cookies. 
+--
+-- The library consists of the following class
+--
+-- * <code>CookieJar</code>
+-- ** This is the main class which holds the cookie jar. 
+--
+-- The following sample code shows how the cookie jar could be used:
+-- This code below will help us to pass cookies to the http calls, receive
+-- further cookies and append them properly to the existing cookie jar.
+-- 
+-- <code>
+--   local cookie_table = {}
+--   table.insert(cookie_table, <optional cookie table>)
+--   local cookiejar = httpcookies.CookieJar:new(cookie_table)
+--   cookiejar:set_no_cookie_overwrite(true)
+--   
+--   local response
+--   response = cookiejar:get(host, port, path, options)
+--
+--   return response
+-- </code>
+--
+-- The following sample command tells us how to make use of the library
+-- in the existing scripts :
+-- 
+-- $nmap --script http-xss-scanner --script-args httpcookies.cookiejar={<table of cookies>}
```

Comment:
Same here. Add an example of <table of cookies>. You haven't mentioned the structure of cookies.